spec/spec_helper.cr

```diff
@@ -15,12 +12,19 @@ def new_test_context(verb = "GET", path = "/api/test")
   {backing_io, HTTP::Server::Context.new(request, response)}
 end
 
-DppmRestApi.run Socket::IPAddress::LOOPBACK, DPPM::Prefix.default_dppm_config.port, __DIR__
+Kemal.config.env = "test"
+
+# Set up the mock permissions.json
 
-# Because the API key for the "normal user" is automatically generated, we
-# need to update the configuration to match the key that was just generated.
-usr = DppmRestApi.permissions_config.users.find { |user| user.name == "Jim Oliver" }.not_nil!
-DppmRestApi.permissions_config.users.delete usr
-usr.api_key_hash = Scrypt::Password.create NORMAL_USER_API_KEY
-DppmRestApi.permissions_config.users << usr
-DppmRestApi.permissions_config.write_to PERMISSION_FILE
+# the location
+PERMISSION_FILE = Path[__DIR__, "permissions.json"]
+
+Spec.before_each do
+  DppmRestApi.permissions_config = Fixtures.permissions_config
+  DppmRestApi.permissions_config.write_to PERMISSION_FILE
+  DppmRestApi.run Socket::IPAddress::LOOPBACK, DPPM::Prefix.default_dppm_config.port, __DIR__
```

Comment:
You think it's better to create a new server for each spec? This will slow it down.
Reseting the permission file may be enough, with eventually reloading the server.

Reply:
Generally, I prefer to make my tests as atomic and robust as possible, with little regard to the time it will take. However, in the case of these changes, you are right -- the slowdown is _significant_. I suspect the majority of that is from the time spent writing, deleting, and rewriting the file, but I'll do some checking and see if your idea will help significantly.

I was thinking earlier, perhaps instead of deleting the file every time, just rewrite it if the MD5 has does not match. This would be a read-only activity which could be cached by the Kernel. I will do some testing.